core/build.gradle

```diff
@@ -1,48 +1,19 @@
-apply plugin: "com.android.library"
+apply from: "$gradle.ext.projectRoot/common_library_config.gradle"
```

Comment:
Could this also be achieved by adding things to the `allprojects` block in the main `build.gradle`? I think that could be a bit more clean because it doesn't need the `gradle.ext.projectRoot` assignment in `settings.gradle`.

Reply:
I do not believe so. I tried messing around with it a bit, and it didn't want to work.

Reply:
The `SpotBugs` setup is also done in the `allprojects` section. I'm pretty sure it is possible to specify it centrally instead of including a file.

> the app module cannot be covered by this, at it is not a library module

When using `allprojects`, you can specify everything except the `apply plugin` line in the root gradle file

Reply:
Just played around with `allprojects` a bit. I eventually got it to work but now I believe you that doing it with `allprojects` is not clean. Would it be possible that each module specifies both, the plugin and the include? Then we can re-use as much code as possible.

```
apply plugin: "com.android.library"
apply from: "$gradle.ext.projectRoot/common_library_config.gradle"
```

Reply:
> Would it be possible that each module specifies both, the plugin and the include? Then we can re-use as much code as possible.

I don't think so, because then you run into the problem of gradle failing because no plugin was specified for common_library_config 🤷 (`Could not find method android()` and things like that). It sucks, but I believe this is about as much as we can re-use.

Reply:
I think that should work if you put the Android plugin before the file include. Then the same can be done to extract the definition of the build flavors. Could you please just call the include file `common.gradle` and also move the `allprojects` parts (except `repositories`) to the `common.gradle` file?

Reply:
Hi @TacoTheDank - Are you still interested in this PR? If not, I will do the changes myself.